### PR TITLE
component/event-video-player

### LINKED ISSUE #66: Meeting video player
## Use Case
*Please provide a use case to help us understand your request in context*
Rewrite video player from seattle repo:
https://github.com/CouncilDataProject/seattle/blob/master/src/components/Event.jsx#L205
image

## Solution
*Please describe your ideal solution*
For now, use [video-react](https://video-react.js.org/), which is currently used in seattle repo, and add a replay control as requested in [#75](https://github.com/CouncilDataProject/seattle/issues/75).

Add replay control documentation: https://video-react.js.org/components/replay-control/

Don't copy over the feature moving the video player to the right side of the page on scroll as most browsers have their own picture-in-picture functionality now.

## Alternatives
*Please describe any alternatives you've considered, even if you've dismissed them*

### PR DESCRIPTION
**Pull request by tohuynh**
**Pull request recommendations:**
- [x] Name your pull request _your-development-type/short-description_. Ex: _feature/read-tiff-files_
- [x] Link to any relevant issue in the PR description. Ex: _Resolves [gh-12], adds tiff file format support_
  Resolves #66 
- [x] Provide context of changes.

- [video-react](https://www.npmjs.com/package/video-react) is no longer being developed. Instead, we'll use [video.js](https://videojs.com/), which video-react was based on.
Unfortunately, video.js doesn't have a skip back/forward X seconds in its UI. For that I added a keyboard event handler to allow users to skip back/forward 10 seconds using arrow left and right, along with other standard keyboard events. `k` and `space` for play/pause. `m` for mute/unmute, and `f` for enter/exit full screen.

- I create a custom theme to style the video player, adapting from https://github.com/videojs/themes. I tried to make it like Youtube with the primary color being cdp_neon_blue.

- The player also supports language localization, by using `strings.getLanuage` and the player will use video.js language file.

- To create a 'jump to setence in video clip` call back for uss in #59 , I created a public api with a seekTo method.
```
const eventVideoRef = createRef<EventVideoRef>();
<EventVideo {..other props} componentRef={eventVideoRef} />
//create jump to video clip callback which then can then be passed to TranscriptFull and TranscriptSearch
const jumpToVideoClip = (startTime: number) => eventVideoRef.current?.seekTo(startTime)
```
- [ ] Provide relevant tests for your feature or bug fix.
- [x] Provide or update documentation for any feature added by your pull request.
https://tohuynh.github.io/cdp-frontend/?path=/story/library-details-event-video--default

Thanks for contributing!

### CODE REVIEW

**Comment by hawkticehurst** [comment]
An immediate piece of feedback before a proper review is that the usage of `cdp_neon_blue` with the white foreground does not have enough contrast to be accessible, so I would pick a different background or foreground color.

**Comment by tohuynh** [comment]
> An immediate piece of feedback before a proper review is that the usage of `cdp_neon_blue` with the white foreground does not have enough contrast to be accessible, so I would pick a different background or foreground color.

The video control component's background-color isn't entirely white, it's a linear gradient of black whose opacity gradually decreases as it moves toward the top of the component. I agree about the low contrast of white and cdp_neon_blue though in the big play button.

File: src/components/Details/EventVideo/EventVideo.tsx [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+            timeDivider: true,
+            durationDisplay: true,
+            customControlSpacer: true,
+            remainingTimeDisplay: false,
+          },
+          preload: initialSeconds ? "auto" : "metadata",
+          aspectRatio: "16:9",
+          fluid: true,
+          language: strings.getLanguage(),
+          playbackRates: [0.75, 1, 1.5, 2],
```
Comment (JacksonMaxfield) [inline src/components/Details/EventVideo/EventVideo.tsx:82]: Can we add 0.5 and reverse the order of this list. I am so used to the youtube version where it's the reverse order of this.

**Review by JacksonMaxfield — APPROVED** [review]
Only one change from me otherwise this is excellent!

File: src/components/Details/EventVideo/EventVideo.tsx (hunk 1 above)
Comment (tohuynh) [reply src/components/Details/EventVideo/EventVideo.tsx:82]: Yes to both

**Pull request merged**